tests/functional/ingestion/IngestionReader.js

```diff
@@ -103,112 +104,66 @@ describe('ingestion reader tests with mock', function fD() {
-            next => kafkaAdminClient.createTopic({
-                    topic,
-                    num_partitions: 1, // eslint-disable-line camelcase
-                    replication_factor: 1, // eslint-disable-line camelcase
-                }, err => {
-                    if (err && err.code === 36) {
-                        // if topic already exits.
-                        return next();
-                    }
-                    return next(err);
-                }),
-            next => {
-                producer = new BackbeatProducer({
-                    kafka: testConfig.kafka,
-                    topic,
-                });
-                producer.once('ready', () => {
-                    next();
-                });
-            },
-            next => {
-                consumer.connect({ timeout: 1000 }, () => { });
-                consumer.once('ready', () => {
-                    next();
-                });
-            },
-            next => {
-                consumer.subscribe([testConfig.extensions.ingestion.topic]);
-                setTimeout(next, 2000);
-            },
-            next => this.db.createCollection('PENSIEVE', err => {
-                assert.ifError(err);
-                return next();
-            }),
-            next => {
-                this.m = this.db.collection('PENSIEVE');
-                this.m.insertOne(dummyPensieveCredentials, {});
-                return next();
-            },
-            next => {
-                this.m.insertOne({
-                    _id: 'configuration/overlay-version',
-                    value: 6,
-                }, {});
-                return next();
-            },
-            next => {
-                this.m.insertOne(dummySSHKey, {});
-                return next();
-            },
-            next => {
-                const cbOnce = jsutil.once(next);
-                zkClient = new ZookeeperManager('localhost:2181', { autoCreateNamespace: true }, dummyLogger);
-                zkClient.once('error', cbOnce);
-                zkClient.once('ready', () => {
-                    zkClient.removeAllListeners('error');
-                    return cbOnce();
-                });
-            },
-            next => initManagement(testConfig, next),
-            next => {
-                const metadataMock = new MetadataMock();
-                httpServer = http.createServer(
-                    (req, res) => metadataMock.onRequest(req, res))
-                    .listen(testPort);
-                next();
+        await client.connect();
+        db = client.db('metadata', { ignoreUndefined: true });
+        try {
+            const createTopic = promisify(kafkaAdminClient.createTopic).bind(kafkaAdminClient);
+            await createTopic({
+                topic,
+                num_partitions: 1, // eslint-disable-line camelcase
+                replication_factor: 1, // eslint-disable-line camelcase
+        }).catch(err => {
```

Comment:
bad indent: this is not try/catch, but the `catch` function of the promise...

instead of mixing try/catch and `Promise.catch()`, may be more consistant to just use try/catch locally (and remove the overall `catch` block which simply rethrows at the end)

```suggestion
        try {
            const createTopic = promisify(kafkaAdminClient.createTopic).bind(kafkaAdminClient);
            await createTopic({ ... });
        } catch(err) {
            if (err.code !== 36) { // if topic does not already exist
                throw err;
            }
        }
```